WatchDogCore/WatchDog/src/nl/tudelft/watchdog/core/logic/interval/intervaltypes/IntervalBase.java

```diff
@@ -122,7 +122,11 @@ public IntervalType getType() {
 
 	/** Necessary for storage of Intervals. */
 	public int compareTo(IntervalBase comparedInterval) {
-		return getEnd().compareTo(comparedInterval.getEnd());
+		int res = getEnd().compareTo(comparedInterval.getEnd());
+		if (res == 0 && !this.equals(comparedInterval)) {
+			res = -1;
+		}
+		return res;
```

Comment:
Yes. In your example a!=b and compareTo(a,b)=-1. This means that compareTo(a.end, b.end)=-1, and therefore a.end!=b.end. This means that compareTo(b.end,a.end)= 1, and therefore also compareTo(b,a)=1, as the return value is only changed to -1 when compareTo(b.end,a.end)=0.